dask/dataframe/io/parquet.py

```diff
@@ -397,22 +394,23 @@ def _write_pyarrow(df, path, write_index=None, append=False,
     template = fs.sep.join([path, 'part.%i.parquet'])
 
     write = delayed(_write_partition_pyarrow)
-    first_kwargs = kwargs.copy()
-    first_kwargs['metadata_path'] = fs.sep.join([path, '_metadata'])
-    writes = [write(part, open_with, template % i, write_index,
-                    **(kwargs if i else first_kwargs))
+    metadata_path = fs.sep.join([path, '_metadata'])
+    writes = [write(part, open_with, template % i, write_index, compression, metadata_path,
+                    **kwargs)
```

Comment:
Please revert the handling of `first_kwargs`/`metadata_path` here - the way you changed it the metadata is written once for every task instead of just once by the first task.

Reply:
done